sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClient.cs

```diff
@@ -134,7 +151,14 @@ public virtual async Task<Response<PersonalizerRankResult>> RankAsync(Personaliz
             scope.Start();
             try
             {
-                return await RankRestClient.RankAsync(options, cancellationToken).ConfigureAwait(false);
+                if (_isLocalInference)
+                {
+                    return _rankProcessor.Rank(options);
```

Comment:
does rankProcessor not have an async method?

Reply:
My concern is that the user is calling RankAsync, but nothing asynchronous is happening in the local inference code path